lib/os/heap-validate.c

```diff
@@ -386,3 +386,61 @@ void sys_heap_print_info(struct sys_heap *heap, bool dump_chunks)
 {
 	heap_print_info(heap->heap, dump_chunks);
 }
+
+#ifdef CONFIG_SYS_HEAP_RUNTIME_STATS
+
+int heap_runtime_stats_get(struct sys_heap *heap,
+		struct sys_heap_runtime_stats *stats)
+{
+	size_t free_bytes = 0;
+	struct z_heap *h;
+
+	if ((heap == NULL) || (stats == NULL)) {
+		return -EINVAL;
+	}
+
+	h = heap->heap;
+	stats->free_bytes = 0;
+	stats->allocated_bytes = 0;
+	stats->max_avail_bytes = 0;
+
+	/* Iterate through all chunks */
+	for (chunkid_t c = 0; ; c = right_chunk(h, c)) {
+		if (chunk_used(h, c)) {
+			if ((c != 0) && (c != h->end_chunk)) {
+				stats->allocated_bytes += chunksz_to_bytes(
+						h, chunk_size(h, c));
+			}
+		} else {
+			if (!solo_free_header(h, c)) {
+				free_bytes = chunksz_to_bytes(h,
+						chunk_size(h, c));
+				stats->free_bytes += free_bytes;
+				if (free_bytes > stats->max_avail_bytes) {
+					stats->max_avail_bytes = free_bytes;
+				}
+			}
+		}
+		if (c == h->end_chunk) {
+			break;
+		}
+	}
```

Comment:
It is better to replace "for(){}" with "do...while()".

do {
} while (c != h->end_chunk)

Reply:
I have updated it, thanks,